File: VB/TestMyTreeListForum/Form1.vb
Public Class FormMain
    Public Sub New()
        InitializeComponent()

        For i As Integer = 0 To myTreeList1.Columns.Count - 1
            If i = 1 Then
                myTreeList1.Columns(i).VisibleIndex = (0)
            Else
                myTreeList1.Columns(i).VisibleIndex = (-1)
            End If
        Next i

        myTreeList1.KeyFieldName = "ID"
        myTreeList1.ParentFieldName = "ParentID"
        myTreeList1.AvatarImageFieldName = "Avatar"
        myTreeList1.AvatarCaptionFieldName = "FirstName"
    End Sub



    Private Sub FormMain_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        'TODO: This line of code loads data into the 'MyDataDataSet.Customers' table. You can move, or remove it, as needed.
        Me.CustomersTableAdapter.Fill(Me.MyDataDataSet.Customers)

    End Sub
End Class

File: CS/TestMyTreeListForum/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraTreeList.Nodes;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.MyControl;



namespace TestMyTreeListForum
{
    public partial class FormMain : Form
    {
        public FormMain()
        {
            InitializeComponent();

            for (int i = 0; i < myTreeList1.Columns.Count; i++)
                myTreeList1.Columns[i].VisibleIndex = (i == 1 ? 0 : -1);

            myTreeList1.KeyFieldName = "ID";
            myTreeList1.ParentFieldName = "ParentID";
            myTreeList1.AvatarImageFieldName = "Avatar";
            myTreeList1.AvatarCaptionFieldName = "FirstName";
            myTreeList1.OptionsView.ShowIndentAsRowStyle = true;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            this.customersTableAdapter.Fill(this.myDataDataSet.Customers);
        }
    }
}